### PR TITLE
Adding loader for TONAS

### PR DESCRIPTION
**Pull request by genisplaja**
### Description

Please include the following information at the top level docstring for the dataset's module mydataset.py:

- [x] Describe annotations included in the dataset
- [x] Indicate the size of the datasets (e.g. number files and duration, hours)
- [x] Mention the origin of the dataset (e.g. creator, institution)
- [x] Describe the type of music included in the dataset
- [x] Indicate any relevant papers related to the dataset
- [x] Include a description about how the data can be accessed and the license it uses (if applicable)

#### Dataset loaders checklist:

- [x] Create a script in `scripts/`, e.g. `make_my_dataset_index.py`, which generates an index file.
- [x] Run the script on the canonical version of the dataset and save the index in `mirdata/indexes/` e.g. `my_dataset_index.json`.
- [x] Create a module in mirdata, e.g. `mirdata/my_dataset.py`
- [x] Create tests for your loader in `tests/`, e.g. `test_my_dataset.py`
- [x] Add your module to `docs/source/mirdata.rst` and `docs/source/quick_reference.rst`
- [x] Run `tests/test_full_dataset.py` on your dataset.

### CODE REVIEW

File: mirdata/datasets/tonas.py [hunk 1]
```diff
@@ -0,0 +1,409 @@
...
+    The annotations are represented by specifying the value (in this case, Notes and F0) at the related timestamps.
+    TONAS' note and F0 annotations also have "Energy" information, which refers to the average energy value through
+    all the frames in which a note or a F0 value is comprised.
+
+    Using this dataset:
+    TONAS dataset can be obtained upon request. Please refer to this link: https://zenodo.org/record/1290722 to
+    request access and follow the indications of the .download() method for a proper storing and organization
+    of the TONAS dataset.
+    When TONAS is used for academic research, we would highly appreciate if scientific publications of works
+    partly based on the TONAS dataset quote the following publication:
```
Comment (magdalenafuentes) [inline mirdata/datasets/tonas.py:32]: There's no publication listed?

File: mirdata/datasets/tonas.py [hunk 2]
```diff
@@ -0,0 +1,409 @@
...
+import os
+from typing import cast, TextIO, Tuple, Optional
+
+import librosa
+import numpy as np
+
+from mirdata import jams_utils
+from mirdata import core
+from mirdata import annotations
+from mirdata import io
```
Comment (magdalenafuentes) [inline mirdata/datasets/tonas.py:45]: import all in one line

File: mirdata/datasets/tonas.py [hunk 3]
```diff
@@ -0,0 +1,409 @@
...
+    author = {E. {Gómez} and J. {Bonada}},
+    journal = {Computer Music Journal},
+    title = {Towards Computer-Assisted Flamenco Transcription: An Experimental 
+           Comparison of Automatic Transcription Algorithms as Applied to A 
+           Cappella Singing},
+    year = {2013},
+    volume = {37},
+    number = {2},
+    pages = {73-90},
+    doi = {10.1162/COMJ_a_00180}}
```
Comment (magdalenafuentes) [inline mirdata/datasets/tonas.py:78]: Is there any instructions on what to cite when? Or should the user cite all the papers every time?

File: mirdata/datasets/tonas.py [hunk 4]
```diff
@@ -0,0 +1,409 @@
...
+"""
+
+LICENSE_INFO = """
+The TONAS dataset is offered free of charge for internal non-commercial use only. You can not redistribute it nor 
+modify it. Dataset by COFLA team. Copyright © 2012 COFLA project, Universidad de Sevilla. Distribution rights granted 
+to Music Technology Group, Universitat Pompeu Fabra. All Rights Reserved.
+"""
+
+# NoteData extension for Tonas to support energy annotation
+class NoteData(annotations.NoteData):
```
Comment (magdalenafuentes) [inline mirdata/datasets/tonas.py:100]: ♥️

File: mirdata/datasets/tonas.py [hunk 5]
```diff
@@ -0,0 +1,409 @@
...
+    freqs_corr = []
+    energies = []
+    reader = np.genfromtxt(fhandle)
+    for line in reader:
+        times.append(float(line[0]))
+        energies.append(float(line[1]))
+        freqs.append(float(line[2]))
+        freqs_corr.append(float(line[3]))
+
+    times = np.array(times, dtype="float")  # type: ignore
```
Comment (magdalenafuentes) [inline mirdata/datasets/tonas.py:277]: What do you mean by `# type: ignore`?

File: mirdata/datasets/tonas.py (hunk 4 above)
Comment (magdalenafuentes) [reply mirdata/datasets/tonas.py:100]: Maybe add a small description on the two extended classes just indicating what you are adding

File: mirdata/datasets/tonas.py [hunk 6]
```diff
@@ -0,0 +1,409 @@
...
+        annotations.validate_confidence(confidence)
+
+        self.intervals = intervals
+        self.notes = notes
+        self.energies = energies
+        self.confidence = confidence
+
+
+# F0 extension for Tonas to support energy annotation
+class F0Data(annotations.F0Data):
```
Comment (magdalenafuentes) [inline mirdata/datasets/tonas.py:121]: Especially in this one that you have the automatically estimated frequencies?

File: mirdata/datasets/tonas.py [hunk 7]
```diff
@@ -0,0 +1,409 @@
...
+        np.array(pitches, dtype="float"),
+        np.array(energy, dtype="float"),
+        np.array(confidence, dtype="float"),
+    )
+
+    return note_data
+
+
+@io.coerce_to_string_io
+def load_tuning_frequency(fhandle: TextIO) -> float:
```
Comment (magdalenafuentes) [inline mirdata/datasets/tonas.py:323]: Is this a private function? if so start with _

File: mirdata/datasets/tonas.py [hunk 8]
```diff
@@ -0,0 +1,409 @@
...
+    # Compute tuning frequency
+    cents_deviation = float(next(csv.reader(fhandle, delimiter=","))[0])
+    tuning_frequency = 440 * (
+        2 ** (cents_deviation / 1200)
+    )  # Frequency of A (common value is 440Hz)
+
+    return tuning_frequency
+
+
+def midi_to_hz(midi_note, tuning_deviation):
```
Comment (magdalenafuentes) [inline mirdata/datasets/tonas.py:343]: Same question as above + would this be equivalent to `librosa`'s [`midi_to_hz`](http://man.hubwiz.com/docset/LibROSA.docset/Contents/Resources/Documents/generated/librosa.core.midi_to_hz.html)?

File: scripts/make_tonas_index.py [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+    Returns
+    -------
+    md5_hash: str
+        md5 hash of data in file_path
+    """
+    hash_md5 = hashlib.md5()
+    with open(file_path, "rb") as fhandle:
+        for chunk in iter(lambda: fhandle.read(4096), b""):
+            hash_md5.update(chunk)
+    return hash_md5.hexdigest()
```
Comment (magdalenafuentes) [inline scripts/make_tonas_index.py:27]: import from validate

File: tests/resources/mir_datasets/tonas/README.txt [hunk 10]
```diff
@@ -0,0 +1,74 @@
+TONAS: COFLA dataset of a cappella flamenco singing
```
Comment (magdalenafuentes) [inline tests/resources/mir_datasets/tonas/README.txt:1]: Do you need this file here? If not remove

**Review by magdalenafuentes — APPROVED** [review]
@genisplaja Great work! ♥️

Just left minor comments and is ready to go. The `wav` file for testing seems a bit big, could  you please trim it?

File: tests/resources/mir_datasets/tonas/README.txt (hunk 10 above)
Comment (genisplaja) [reply tests/resources/mir_datasets/tonas/README.txt]: Done!

File: mirdata/datasets/tonas.py (hunk 6 above)
Comment (genisplaja) [reply mirdata/datasets/tonas.py:132]: Yes! I have updates de docstring for both extended classes, and I have specified this matter. F0 annotations have the automatically extracted F0 and then the semi-automatic F0 as well.

File: mirdata/datasets/tonas.py (hunk 4 above)
Comment (genisplaja) [reply mirdata/datasets/tonas.py:98]: Added docstrings for both extended classes!:)

File: mirdata/datasets/tonas.py (hunk 3 above)
Comment (genisplaja) [reply mirdata/datasets/tonas.py]: Fixed. I just copied and pasted the related citations here, but now I have put the needed ones with a little comment to what they exactly refer.

File: mirdata/datasets/tonas.py (hunk 1 above)
Comment (genisplaja) [reply mirdata/datasets/tonas.py]: Missed it. Now is there:)

File: mirdata/datasets/tonas.py (hunk 8 above)
Comment (genisplaja) [reply mirdata/datasets/tonas.py]: hmmm, the thing is, I wrote my function here to easily support the fact that the tunning freq is different from 440, and the info we get from TONAS about this new tunning freq is the deviation is cents from 440. I wrote the function to adapt to this data organization!

**Comment by genisplaja** [comment]
Thank you very much @magdalenafuentes for your comments and sorry for the delay in addressing them! I commented on some reviews, but for most of them, I corrected the related issues already. Ah, and I trimmed the test track: it was about 30 seconds (my bad to forget!) and now it's 1 second. I'll update now the branch.

**Comment by genisplaja** [comment]
I don't know why docs are failing after updating the branch:( I got into the bug details, but I couldn't find any error description. Am I missing something? Thanks!

File: mirdata/datasets/tonas.py (hunk 5 above)
Comment (genisplaja) [reply mirdata/datasets/tonas.py]: Nothing sorry! Copied, pasted, and adapted the function from another loader, and forgot to remove these comments!

File: mirdata/core.py [hunk 11]
```diff
@@ -188,6 +188,9 @@ def default_path(self):
 
         """
         mir_datasets_dir = os.path.join(os.getenv("HOME", "/tmp"), "mir_datasets")
+        # Adapt dataset default path for TONAS
+        if self.name == "tonas":
+            self.name = "TONAS"
```
Comment (rabitt) [inline mirdata/core.py:193]: Why do we need this? I guess the original data is in a folder called `TONAS`? For me this should just be part of the download instructions - we shouldn't put special cases for datasets within `core.py`, it should stay as generic as possible. (In short, I think these three lines should be removed)

File: mirdata/datasets/tonas.py [hunk 12]
```diff
@@ -0,0 +1,438 @@
...
+            confidence, np.ndarray, float, none_allowed=True
+        )
+        annotations.validate_lengths_equal([intervals, notes, energies, confidence])
+        annotations.validate_intervals(intervals)
+        annotations.validate_confidence(confidence)
+
+        self.intervals = intervals
+        self.notes = notes
+        self.energies = energies
+        self.confidence = confidence
```
Comment (rabitt) [inline mirdata/datasets/tonas.py:129]: the lines that are part of the __init__ in the base class don't need to be called here - they get called already in line 114!

This whole init can become, equivalently:
```
super().__init__(intervals, notes, confidence)
annotations.validate_array_like(energies, np.ndarray, float)
annotations.validate_lengths_equal([intervals, notes, energies, confidence])
self.energies = energies
```

File: mirdata/datasets/tonas.py (hunk 4 above)
Comment (rabitt) [reply mirdata/datasets/tonas.py:98]: Can we open an issue to remind us to remove these two classes once we extend the base classes to include units?

File: mirdata/datasets/tonas.py [hunk 13]
```diff
@@ -0,0 +1,438 @@
...
+            [times, automatic_frequencies, frequencies, energies, confidence]
+        )
+        annotations.validate_times(times)
+        annotations.validate_confidence(confidence)
+
+        self.times = times
+        self.automatic_frequencies = automatic_frequencies
+        self.frequencies = frequencies
+        self.energies = energies
+        self.confidence = confidence
```
Comment (rabitt) [inline mirdata/datasets/tonas.py:171]: same comment here

**Review by rabitt — CHANGES_REQUESTED** [review]
Sorry for the last minute comments @genisplaja . Just two small things and lets merge this!

**Comment by genisplaja** [comment]
> Sorry for the last minute comments @genisplaja . Just two small things and lets merge this!

No worries! Thank you very much for the feedback:) Will address it ASAP.

File: mirdata/datasets/tonas.py (hunk 12 above)
Comment (genisplaja) [reply mirdata/datasets/tonas.py:129]: oops! You're right. Fixed now!

**Comment by genisplaja** [comment]
I think this is ready to merge! Thank you everyone for your reviews and comments:) Please let me know if you have additional suggestions before merging.

**Review by rabitt — APPROVED** [review]

**Comment by rabitt** [comment]
Looks great, merging!

**Pull request merged**